### DISCUSSION TITLE
Add range support for Vectors in editor

### DISCUSSION THREAD

**Discussion by Sauermann** [godotengine/godot-proposals#6260] [Editor]
Currently `PROPERTY_HINT_RANGE` only works for `int` and `float`.
It would be nice to have this functionality also for vectors.

An implementation example can be seen here (was initially intended as a now superseded bugfix):
https://github.com/Sauermann/godot/tree/proposal-editor-vector-greater-lesser

Creating this as a discussion, because I don't have an actual use-case.

**Comment by Calinou** [comment]
> Creating this as a discussion, because I don't have an actual use-case.

There are many concrete use cases for built-in node properties, such as Viewport's size, TileMap's tile size, etc. It's better to have the editor clamp the value without spamming errors when you try to drag the size below 0 pixels :slightly_smiling_face:

**Reply by Sauermann** [reply]
The Viewport size problem has been solved in godotengine/godot#73000.

**Comment by Sauermann** [comment]
With the recent merge of godotengine/godot#77135, implementing this will become more complex, because changing a value now might move another value beyond its limits.
